src/main/CommandHandler.cpp

```diff
@@ -652,17 +663,83 @@ CommandHandler::ll(std::string const& params, std::string& retStr)
     retStr = root.toStyledString();
 }
 
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+void
+CommandHandler::preflight(std::string const& params, std::string& retStr)
+{
+    ZoneScoped;
+    Json::Value root;
+
+    std::map<std::string, std::string> paramMap;
+    http::server::server::parseParams(params, paramMap);
+    std::string blob = paramMap["blob"];
+    if (!blob.empty())
+    {
+        InvokeHostFunctionOp op;
+        std::vector<uint8_t> binBlob;
+        fromOpaqueBase64(op, blob);
+        root = InvokeHostFunctionOpFrame::preflight(mApp, op);
+    }
+    else
+    {
+        throw std::invalid_argument(
+            "Must specify an InvokeHostFunctionOp blob: preflight?blob=<op in "
+            "base64 XDR format>");
+    }
+    retStr = root.toStyledString();
```

Comment:
`toStyledString` is not the right writer: this one produces a human readable string (with tabs etc), the way to do it is `Json::FastWriter().write(root)`

Reply:
applies to all changes you made in this file, but not all endpoints (for example, the `quorum`, `info` endpoints are commonly used by operators and are not performance sensitive)

Reply:
Fixed.